docs/running-on-kubernetes.md

```diff
@@ -199,9 +226,23 @@ from the other deployment modes. See the [configuration page](configuration.html
   <td><code>false</code></td>
   <td>
     Whether to expose the driver Web UI port as a service NodePort. Turned off by default because NodePort is a limited
-    resource. Use alternatives such as Ingress if possible.
+    resource.
   </td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.driver.serviceManagerType</code></td>
+  <td><code>NodePort</code></td>
+  <td>
+    A tag indicating which class to use for creating the Kubernetes service and determining its URI for the submission
+    client. By default, a service is created with the NodePort type, and the driver will be contacted at one of the
+    kubelet nodes at the port that the Kubelets expose for the service. If the Kubelets cannot be contacted from the
+    submitter's machine, consider setting this to <code>ExternalAnnotation</code> as described in "Determining the
+    Driver Base URI" above. One may also include a custom implementation of
+    <code>org.apache.spark.deploy.rest.kubernetes.DriverServiceManager</code> on the submitter's classpath -
```

Comment:
The advantage of this is that the configuration value becomes easier to set for the things we bundle into it - `NodePort` vs. `ExternalAnnotation` and eventually `ApiServerProxy` is more elegant to remember than full class names.